### ISSUE TITLE
False-positive while_immutable_condition lint when there's braces in the conditional

### ISSUE THREAD

**Issue by mtak-** [rust-lang-nursery/rust-clippy#2596]
Extra braces causes the `while_immutable_condition` lint to trigger incorrectly. The real world code uses `unsafe {}` in the while loop condition, which has the same effect.

```rust
pub fn count_to_0(u: &mut isize) {
    while { *u < 0 } {
        *u += 1;
    }
}
```

```
error: Variable in the condition are not mutated in the loop body. This either leads to an infinite or to a never running loop.
  --> src/epoch.rs:10:11
   |
10 |     while { *u < 0 } {
   |           ^^^^^^^^^^
   |
   = note: #[deny(while_immutable_condition)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.191/index.html#while_immutable_condition
```

Possibly this is a dupe of #2584

**Comment by oli-obk** [comment]
We're ignoring blocks: https://github.com/rust-lang-nursery/rust-clippy/blob/a47734c41d5ebc73bf0b457cf8f9075841132c28/clippy_lints/src/loops.rs#L2227

**Comment by Nokel81** [comment]
I would like to have a go at this
